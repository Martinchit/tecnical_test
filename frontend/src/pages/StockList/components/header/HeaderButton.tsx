import styled from 'styled-components';
import { Button } from 'react-bootstrap';

export const HeaderButton = styled(Button)`
  margin-left: 1%;
  width: 10%;
  min-width: 85px;

  @media (max-width: 767px) {
    width: 50%;
    margin-top: 5%;
    font-size: 0.8rem;
  }

  &:focus {
    outline: none;
    box-shadow: none;
  }
`;
