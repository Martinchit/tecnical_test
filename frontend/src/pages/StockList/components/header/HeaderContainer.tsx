import styled from 'styled-components';
import { FlexBox } from '../../../../core/components/atoms/FlexBox';

export const HeaderContainer = styled(FlexBox)`
  justify-content: flex-end;
  margin: 1% 0;
  width: 100%;

  @media (max-width: 767px) {
    flex
    justify-content: center;
    flex-direction: column;
    margin: 5% 0;
  }
`;
