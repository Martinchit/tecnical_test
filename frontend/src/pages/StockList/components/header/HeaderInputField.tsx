import styled from 'styled-components';
import { FormControl } from 'react-bootstrap';

export const HeaderInputField = styled(FormControl)`
  width: 30%;
  min-width: 250px;

  @media (max-width: 767px) {
    width: 100%;
  }
`;
