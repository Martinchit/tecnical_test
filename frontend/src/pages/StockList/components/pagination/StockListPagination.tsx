import React from 'react';
import { Pagination } from 'react-bootstrap';

interface StockListPaginationProps {
  currentPage: number;
  totalPages: number;
  onPaginationButtonClickHandler: (page: number) => void;
}

export const StockListPagination: React.FC<StockListPaginationProps> = ({
  currentPage,
  totalPages,
  onPaginationButtonClickHandler,
}) => {
  const ref = currentPage + 1 >= totalPages ? totalPages - currentPage + 3 : 5;
  const ellipsisMin = currentPage - 5 < 1 ? 1 : currentPage - 5;
  const ellipsisMax = currentPage + 5 > totalPages ? totalPages : currentPage + 5;
  return (
    <Pagination>
      <Pagination.First onClick={() => onPaginationButtonClickHandler(1)} />
      {currentPage > 3 ? (
        <>
          <Pagination.Item onClick={() => onPaginationButtonClickHandler(1)}>
            {1}
          </Pagination.Item>
          <Pagination.Ellipsis
            onClick={() => onPaginationButtonClickHandler(ellipsisMin)}
          />
        </>
      ) : null}
      {totalPages > 4
        ? [...Array(ref)].map((_, i) => {
            const p = currentPage < 3 ? 1 + i : currentPage + (-2 + i);
            return (
              <Pagination.Item
                onClick={() => onPaginationButtonClickHandler(p)}
                key={p}
                active={currentPage === p}
              >
                {p}
              </Pagination.Item>
            );
          })
        : [...Array(totalPages)].map((_, i) => {
            const p = currentPage < 3 ? 1 + i : currentPage + (-2 + i);
            return (
              <Pagination.Item
                onClick={() => onPaginationButtonClickHandler(p)}
                key={p}
                active={currentPage === p}
              >
                {p}
              </Pagination.Item>
            );
          })}
      {currentPage < totalPages - 2 && totalPages > 5 ? (
        <>
          <Pagination.Ellipsis
            onClick={() => onPaginationButtonClickHandler(ellipsisMax)}
          />
          <Pagination.Item
            onClick={() => onPaginationButtonClickHandler(totalPages)}
          >
            {totalPages}
          </Pagination.Item>
        </>
      ) : null}
      <Pagination.Last onClick={() => onPaginationButtonClickHandler(totalPages)} />
    </Pagination>
  );
};
