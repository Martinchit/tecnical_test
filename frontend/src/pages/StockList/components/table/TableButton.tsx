import styled from 'styled-components';
import { Button } from 'react-bootstrap';

export const TableButton = styled(Button)`
  margin-right: 3%;
  width: 40%;

  @media (max-width: 767px) {
    width: 100%;
    margin-top: 5%;
    font-size: 0.8rem;
  }

  &:focus {
    outline: none;
    box-shadow: none;
  }
`;
