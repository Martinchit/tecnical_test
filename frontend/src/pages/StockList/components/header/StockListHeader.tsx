import React from 'react';
import { HeaderContainer } from './HeaderContainer';
import { HeaderInputField } from './HeaderInputField';
import { HeaderButton } from './HeaderButton';

interface StockListHeaderProps {
  onSearchValueChangeHandler: (value: string) => void;
  onSearchButtonClickHandler: () => void;
  onClearButtonClickHandler: () => void;
  searchValue: string;
  buttonsDisable: boolean;
  isStocksFiltered: boolean;
}

export const StockListHeader: React.FC<StockListHeaderProps> = ({
  onClearButtonClickHandler,
  onSearchValueChangeHandler,
  onSearchButtonClickHandler,
  searchValue,
  buttonsDisable,
  isStocksFiltered,
}) => {
  return (
    <HeaderContainer>
      <HeaderInputField
        value={searchValue}
        onChange={({ target }) => onSearchValueChangeHandler(target.value)}
        placeholder="Search Stock by Stock Code"
      />
      <HeaderButton
        disabled={buttonsDisable}
        variant="primary"
        onClick={onSearchButtonClickHandler}
      >
        Search
      </HeaderButton>
      <HeaderButton
        disabled={buttonsDisable || !isStocksFiltered}
        variant="warning"
        onClick={onClearButtonClickHandler}
      >
        Clear
      </HeaderButton>
    </HeaderContainer>
  );
};
