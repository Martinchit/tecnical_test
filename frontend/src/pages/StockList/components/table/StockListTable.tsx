import React from 'react';
import { Table } from 'react-bootstrap';
import { TableButton } from './TableButton';
import { TableErrorContainer } from '../../../../core/components/atoms/TableErrorContainer';
import { TableCell } from '../../../../core/components/atoms/TableCell';
import { TableHead } from '../../../../core/components/atoms/TableHead';
import { Alert } from '../../../../core/components/atoms/Alert';
import { OrderSide, Stock, StockListAlert } from '../../../../types/types';

interface StockListTableProps {
  alert: StockListAlert;
  stocks: Stock[];
  onActionButtonClickHandler: (action: OrderSide, idx: number) => void;
}

export const StockListTable: React.FC<StockListTableProps> = ({
  alert,
  stocks,
  onActionButtonClickHandler,
}) => {
  return (
    <Table responsive>
      <thead>
        <tr>
          <TableHead>Stock Code</TableHead>
          <TableHead>Market Price</TableHead>
          <TableHead>Currency</TableHead>
          <TableHead>Actions</TableHead>
        </tr>
      </thead>
      <tbody>
        {stocks.map(({ id, currency, price, bloombergTickerLocal }, idx) => (
          <React.Fragment key={bloombergTickerLocal}>
            <tr>
              <TableCell>{bloombergTickerLocal}</TableCell>
              <TableCell>{price}</TableCell>
              <TableCell>{currency}</TableCell>
              <TableCell>
                <TableButton
                  variant="primary"
                  onClick={() => onActionButtonClickHandler('Buy', idx)}
                >
                  Buy
                </TableButton>
                <TableButton
                  variant="secondary"
                  onClick={() => onActionButtonClickHandler('Sell', idx)}
                >
                  Sell
                </TableButton>
              </TableCell>
            </tr>
            {alert[id] ? (
              <tr>
                <TableErrorContainer colSpan={4}>
                  <Alert variant="success">
                    {`${alert[id]} order for ${bloombergTickerLocal} is added to the Orders Basket`}
                  </Alert>
                </TableErrorContainer>
              </tr>
            ) : null}
          </React.Fragment>
        ))}
      </tbody>
    </Table>
  );
};
