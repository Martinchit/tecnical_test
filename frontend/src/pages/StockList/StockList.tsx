import React from 'react';
import axios from 'axios';
import { v4 as uuidv4 } from 'uuid';
import { StockListHeader } from './components/header/StockListHeader';
import { StockListTable } from './components/table/StockListTable';
import { StockListPagination } from './components/pagination/StockListPagination';
import { PageTitle } from '../../core/components/molecules/PageTitle';
import { FlexBox } from '../../core/components/atoms/FlexBox';
import { Spinner } from '../../core/components/molecules/Spinner';
import { Alert } from '../../core/components/atoms/Alert';
import {
  OrderSide,
  Stock,
  StockOrder,
  StockListAlert,
} from '../../types/types';

interface StockListState {
  currentPage: number;
  totalPages: number;
  getTotalPagesLoading: boolean;
  getTotalPagesError: boolean;
  alert: StockListAlert;
  searchValue: string;
  stocksFiltered: boolean;
}

interface StockListProps {
  getStocksAction: (page: number, token: string) => void;
  addStockInOrderAction: (stockOrder: StockOrder) => void;
  searchStocksAction: (query: string, page: number, token: string) => void;
  logOutAction: () => void;
  stocks: Stock[];
  loading: boolean;
  error: boolean;
  token: string;
  sessionTimeout: boolean;
}

export class StockList extends React.Component<StockListProps, StockListState> {
  constructor(props: StockListProps) {
    super(props);
    this.state = {
      currentPage: 1,
      totalPages: 0,
      getTotalPagesLoading: false,
      getTotalPagesError: false,
      alert: {},
      searchValue: '',
      stocksFiltered: false,
    };
  }

  componentDidMount = () => {
    this.getStocksTotalPages();
  };

  componentDidUpdate = () => {
    const { sessionTimeout, logOutAction } = this.props;
    if (sessionTimeout) {
      alert('Session Timeout - Please log in again!');
      logOutAction();
    }
  };

  getStocksTotalPages = (query: string | null = null) => {
    const { token } = this.props;
    this.setState({ getTotalPagesLoading: true, currentPage: 1 });
    const params = query ? { query } : null;
    const headers = {
      Authorization: `Bearer ${token}`,
    };
    axios({
      method: 'GET',
      baseURL: `${process.env.REACT_APP_API_ENDPOINT}`,
      url: '/stock/total_pages',
      params,
      headers,
    })
      .then((res) => {
        const { totalPages } = res.data;
        const { getStocksAction, searchStocksAction } = this.props;
        const { currentPage } = this.state;
        this.setState({
          totalPages: totalPages + 1,
          getTotalPagesLoading: false,
        });
        if (query) {
          this.setState({ stocksFiltered: true });
          searchStocksAction(query, currentPage, token);
        } else {
          this.setState({ stocksFiltered: false });
          getStocksAction(currentPage, token);
        }
      })
      .catch((errorRes) => {
        const { error } = errorRes;
        console.error(error);
        this.setState({
          getTotalPagesError: true,
          getTotalPagesLoading: false,
        });
      });
  };

  onActionButtonClickHandler = (side: OrderSide, idx: number) => {
    const { addStockInOrderAction, stocks } = this.props;
    const targetedStock = stocks[idx];
    const { id, stockId, currency, bloombergTickerLocal } = targetedStock;
    const alert = { [id]: side };
    const stockOrder: StockOrder = {
      id,
      stockId,
      currency,
      bloombergTickerLocal,
      side,
      status: 'Not Ready',
      selected: false,
      orderId: uuidv4(),
    };
    addStockInOrderAction(stockOrder);
    this.setState({ alert });
  };

  onPaginationButtonClickHandler = (selectedPage: number) => {
    const { stocksFiltered, searchValue } = this.state;
    const { getStocksAction, token, searchStocksAction } = this.props;
    this.setState({ currentPage: selectedPage }, () => {
      if (stocksFiltered) {
        searchStocksAction(searchValue, selectedPage, token);
      } else {
        getStocksAction(selectedPage, token);
      }
    });
  };

  onSearchValueChangeHandler = (searchValue: string) => {
    this.setState({ searchValue });
  };

  onSearchButtonClickHandler = () => {
    const { searchValue } = this.state;
    this.setState({ alert: {} }, () => {
      this.getStocksTotalPages(searchValue);
    });
  };

  onClearButtonClickHandler = () => {
    this.setState({ searchValue: '', alert: {} }, () => {
      this.getStocksTotalPages();
    });
  };

  render() {
    const {
      alert,
      currentPage,
      totalPages,
      getTotalPagesLoading,
      getTotalPagesError,
      searchValue,
      stocksFiltered,
    } = this.state;
    const { loading, stocks } = this.props;
    return (
      <>
        <PageTitle title="Stock List" />
        <StockListHeader
          searchValue={searchValue}
          buttonsDisable={loading || !searchValue.trim().length}
          onSearchValueChangeHandler={this.onSearchValueChangeHandler}
          onSearchButtonClickHandler={this.onSearchButtonClickHandler}
          onClearButtonClickHandler={this.onClearButtonClickHandler}
          isStocksFiltered={stocksFiltered}
        />
        <FlexBox margin="2% 0%">
          {loading || getTotalPagesLoading ? (
            <Spinner />
          ) : getTotalPagesError ? (
            <Alert variant="danger">Network error. Please refresh!</Alert>
          ) : (
            <>
              <StockListTable
                alert={alert}
                stocks={stocks}
                onActionButtonClickHandler={this.onActionButtonClickHandler}
              />
              <FlexBox margin="4% 0 0 0">
                <StockListPagination
                  currentPage={currentPage}
                  totalPages={totalPages}
                  onPaginationButtonClickHandler={
                    this.onPaginationButtonClickHandler
                  }
                />
              </FlexBox>
            </>
          )}
        </FlexBox>
      </>
    );
  }
}
