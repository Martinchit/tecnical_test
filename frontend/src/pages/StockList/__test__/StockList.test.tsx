import React from 'react';
import { shallow } from 'enzyme';
import axios, { AxiosStatic } from 'axios';

import { StockList } from '../StockList';
import { Spinner } from '../../../core/components/molecules/Spinner';
import { Alert } from '../../../core/components/atoms/Alert';
import { StockListHeader } from '../components/header/StockListHeader';
import { StockListPagination } from '../components/pagination/StockListPagination';
import { StockListTable } from '../components/table/StockListTable';
import { OrderSide } from '../../../types/types';

interface AxiosMock extends AxiosStatic {
  mockImplementation: Function;
  mockRestore: Function;
}

const testUUID = '9cc4ca45-b42b-462d-a8aa-931e6041cfcb';

jest.mock('axios');
jest.mock('uuid', () => ({ v4: () => testUUID }));

describe('StockList', () => {
  const totalPages = 10;
  const props = {
    getStocksAction: jest.fn(),
    addStockInOrderAction: jest.fn(),
    searchStocksAction: jest.fn(),
    logOutAction: jest.fn(),
    stocks: [
      {
        id: 1,
        stockId: 'testOne',
        currency: 'HKD',
        ric: 'testOne',
        bloombergTicker: 'testOne',
        bloombergTickerLocal: 'testOne',
        name: 'testOne',
        country: 'testOne',
        price: 100,
      },
      {
        id: 2,
        stockId: 'testTwo',
        currency: 'HKD',
        ric: 'testTwo',
        bloombergTicker: 'testTwo',
        bloombergTickerLocal: 'testTwo',
        name: 'testTwo',
        country: 'testTwo',
        price: 200,
      },
    ],
    loading: false,
    error: false,
    token: 'testToken',
    sessionTimeout: false,
  };
  const waitForAsync = () => new Promise((resolve) => setImmediate(resolve));

  const restoreMock = () => {
    props.getStocksAction.mockRestore();
    props.addStockInOrderAction.mockRestore();
    props.searchStocksAction.mockRestore();
  };

  it('renders correctly', () => {
    const mockedAxios = axios as AxiosMock;
    mockedAxios.mockImplementation(() => {
      return Promise.resolve({
        data: {
          totalPages,
        },
      });
    });
    const wrapper = shallow(<StockList {...props} />);
    expect(wrapper).toMatchSnapshot();
    mockedAxios.mockRestore();
  });

  it('renders Spinner', () => {
    const mockedAxios = axios as AxiosMock;
    mockedAxios.mockImplementation(() => {
      return Promise.resolve({
        data: {
          totalPages,
        },
      });
    });
    const wrapper = shallow(<StockList {...props} />);
    wrapper.setState({ loading: true });
    expect(wrapper.find(Spinner)).toHaveLength(1);
    restoreMock();
    mockedAxios.mockRestore();
  });

  it('calls getStocksTotalPages in componentDidMount', async () => {
    const mockedAxios = axios as AxiosMock;
    mockedAxios.mockImplementation(() => {
      return Promise.resolve({
        data: {
          totalPages,
        },
      });
    });
    const wrapper = shallow(<StockList {...props} />);
    let stockListWrapper = wrapper.instance() as StockList;
    expect(stockListWrapper.state.getTotalPagesLoading).toBeTruthy();
    expect(stockListWrapper.state.currentPage).toBe(1);
    await waitForAsync();
    stockListWrapper = wrapper.instance() as StockList;
    expect(stockListWrapper.state.getTotalPagesLoading).not.toBeTruthy();
    expect(stockListWrapper.state.totalPages).toBe(totalPages + 1);
    expect(stockListWrapper.state.stocksFiltered).not.toBeTruthy();
    expect(props.getStocksAction).toHaveBeenCalled();
    restoreMock();
    mockedAxios.mockRestore();
  });

  it('calls handles getStocksTotalPages error in componentDidMount', async () => {
    const mockedAxios = axios as AxiosMock;
    mockedAxios.mockImplementation(() => {
      // eslint-disable-next-line prefer-promise-reject-errors
      return Promise.reject({ error: new Error('Test Error') });
    });
    const wrapper = shallow(<StockList {...props} />);
    let stockListWrapper = wrapper.instance() as StockList;
    expect(stockListWrapper.state.getTotalPagesLoading).toBeTruthy();
    expect(stockListWrapper.state.currentPage).toBe(1);
    await waitForAsync();
    stockListWrapper = wrapper.instance() as StockList;
    expect(stockListWrapper.state.getTotalPagesLoading).not.toBeTruthy();
    expect(stockListWrapper.state.getTotalPagesError).toBeTruthy();
    await waitForAsync();
    expect(wrapper.find(Alert)).toHaveLength(1);
    restoreMock();
    mockedAxios.mockRestore();
  });

  it('handles dom events in StockListHeader', async () => {
    const searchValue = 'searchTest';
    const mockedAxios = axios as AxiosMock;
    mockedAxios.mockImplementation(() => {
      return Promise.resolve({
        data: {
          totalPages,
        },
      });
    });
    const wrapper = shallow(<StockList {...props} />);
    expect(mockedAxios).toHaveBeenCalledTimes(1);
    await waitForAsync();
    let stockListWrapper = wrapper.instance() as StockList;
    wrapper.find(StockListHeader).props().onSearchValueChangeHandler(searchValue);
    expect(stockListWrapper.state.searchValue).toBe(searchValue);
    await waitForAsync();
    wrapper.find(StockListHeader).props().onSearchButtonClickHandler();
    expect(mockedAxios).toHaveBeenCalledTimes(2);
    wrapper.find(StockListHeader).props().onClearButtonClickHandler();
    stockListWrapper = wrapper.instance() as StockList;
    expect(stockListWrapper.state.searchValue).toBe('');
    expect(mockedAxios).toHaveBeenCalledTimes(3);
    restoreMock();
    mockedAxios.mockRestore();
  });

  it('handles onPaginationButtonClickHandler', async () => {
    const selectedPage = 2;
    const mockedAxios = axios as AxiosMock;
    mockedAxios.mockImplementation(() => {
      return Promise.resolve({
        data: {
          totalPages,
        },
      });
    });
    const wrapper = shallow(<StockList {...props} />);
    expect(mockedAxios).toHaveBeenCalledTimes(1);
    await waitForAsync();
    const stockListWrapper = wrapper.instance() as StockList;
    wrapper.find(StockListPagination).props().onPaginationButtonClickHandler(2);
    expect(stockListWrapper.state.currentPage).toBe(selectedPage);
    expect(props.getStocksAction).toHaveBeenCalled();
    expect(props.getStocksAction).toHaveBeenCalledWith(selectedPage, props.token);
    restoreMock();
    mockedAxios.mockRestore();
  });

  it('handles onPaginationButtonClickHandler in search mode', async () => {
    const selectedPage = 2;
    const mockedAxios = axios as AxiosMock;
    mockedAxios.mockImplementation(() => {
      return Promise.resolve({
        data: {
          totalPages,
        },
      });
    });
    const wrapper = shallow(<StockList {...props} />);
    expect(mockedAxios).toHaveBeenCalledTimes(1);
    await waitForAsync();
    wrapper.setState({ stocksFiltered: true, searchValue: 'testSearch' })
    const stockListWrapper = wrapper.instance() as StockList;
    wrapper.find(StockListPagination).props().onPaginationButtonClickHandler(2);
    expect(stockListWrapper.state.currentPage).toBe(selectedPage);
    expect(props.searchStocksAction).toHaveBeenCalled();
    expect(props.searchStocksAction).toHaveBeenCalledWith(stockListWrapper.state.searchValue, selectedPage, props.token);
    restoreMock();
    mockedAxios.mockRestore();
  });

  it('handles onActionButtonClickHandler', async () => {
    const side = 'Sell' as OrderSide;
    const idx = 0;
    const mockedAxios = axios as AxiosMock;
    mockedAxios.mockImplementation(() => {
      return Promise.resolve({
        data: {
          totalPages,
        },
      });
    });
    const wrapper = shallow(<StockList {...props} />);
    expect(mockedAxios).toHaveBeenCalledTimes(1);
    await waitForAsync();
    wrapper.find(StockListTable).props().onActionButtonClickHandler(side, idx);
    expect(props.addStockInOrderAction).toHaveBeenCalled();
    expect(props.addStockInOrderAction).toHaveBeenCalledWith({
      id: props.stocks[0].id,
      stockId: props.stocks[0].stockId,
      currency: props.stocks[0].currency,
      bloombergTickerLocal: props.stocks[0].bloombergTickerLocal,
      side,
      status: 'Not Ready',
      selected: false,
      orderId: testUUID,
    });
    await waitForAsync();
    const stockListWrapper = wrapper.instance() as StockList;
    expect(stockListWrapper.state.alert).toStrictEqual({
      [props.stocks[0].id]: side,
    });
    restoreMock();
    mockedAxios.mockRestore();
  });

  it('handles sessionTimeout change', () => {
    const mockedAxios = axios as AxiosMock;
    mockedAxios.mockImplementation(() => {
      return Promise.resolve({
        data: {
          totalPages,
        },
      });
    });
    const { alert } = window;
    delete window.alert;
    window.alert = jest.fn();
    const wrapper = shallow(<StockList {...props} />);
    wrapper.setProps({ sessionTimeout: true });
    expect(window.alert).toHaveBeenCalled();
    expect(props.logOutAction).toHaveBeenCalled();
    restoreMock();
    window.alert = alert;
  });
});
