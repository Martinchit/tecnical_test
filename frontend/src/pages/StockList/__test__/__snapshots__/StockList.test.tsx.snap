// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StockList renders correctly 1`] = `
<Fragment>
  <Component
    title="Stock List"
  />
  <Component
    buttonsDisable={true}
    isStocksFiltered={false}
    onClearButtonClickHandler={[Function]}
    onSearchButtonClickHandler={[Function]}
    onSearchValueChangeHandler={[Function]}
    searchValue=""
  />
  <styled.div
    margin="2% 0%"
  >
    <Component />
  </styled.div>
</Fragment>
`;
